source/common/socket.cc

```diff
@@ -0,0 +1,95 @@
+#include "common/socket.h"
+
+#include "external/envoy/source/common/config/utility.h"
+
+namespace Nighthawk {
+
+class SocketConfigFactoryImpl : public UpstreamSocketConfigFactory {};
+
+Envoy::ProtobufTypes::MessagePtr UpstreamSocketConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::transport_socket::TransportSocket>();
+}
+
+Envoy::Network::TransportSocketFactoryPtr UpstreamSocketConfigFactory::createTransportSocketFactory(
+    const Envoy::Protobuf::Message& message,
+    Envoy::Server::Configuration::TransportSocketFactoryContext& context) {
+  const auto& outer_config =
+      Envoy::MessageUtil::downcastAndValidate<const nighthawk::transport_socket::TransportSocket&>(
+          message, context.messageValidationVisitor());
+  auto& inner_config_factory = Envoy::Config::Utility::getAndCheckFactory<
+      Envoy::Server::Configuration::UpstreamTransportSocketConfigFactory>(
+      outer_config.transport_socket());
+  Envoy::ProtobufTypes::MessagePtr inner_factory_config =
+      Envoy::Config::Utility::translateToFactoryConfig(outer_config.transport_socket(),
+                                                       context.messageValidationVisitor(),
+                                                       inner_config_factory);
+  auto inner_transport_factory =
+      inner_config_factory.createTransportSocketFactory(*inner_factory_config, context);
+  return std::make_unique<SocketFactory>(outer_config, SocketConfigFactoryImpl(),
+                                         context.scope().createScope("upstream_socket."),
+                                         std::move(inner_transport_factory));
+}
+
+SocketFactory::SocketFactory(const nighthawk::transport_socket::TransportSocket&,
+                             SocketConfigFactory&&, Envoy::Stats::ScopePtr&& scope,
```

Comment:
Are `const nighthawk::transport_socket::TransportSocket&` and `SocketConfigFactory&&` unused? Can we just remove these parameters? If they are required or used, I didn't understand how.